caffe2/python/convnet_benchmarks.py

```diff
@@ -80,25 +82,32 @@ def MLP(order, cudnn_ws):
                 dim_out=d,
                 weight_init=('XavierFill', {}),
                 bias_init=('XavierFill', {}),
+                WeightInitializer=initializer,
+                BiasInitializer=initializer
             )
     brew.sum(
         model, ["fc_{}_{}".format(depth, j) for j in range(width)], ["sum"]
     )
-    brew.fc(
+    last = brew.fc(
         model,
         "sum",
         "last",
         dim_in=d,
         dim_out=1000,
         weight_init=('XavierFill', {}),
         bias_init=('XavierFill', {}),
+        WeightInitializer=initializer,
+        BiasInitializer=initializer
     )
-    xent = model.net.LabelCrossEntropy(["last", "label"], "xent")
+    if use_fp16:
```

Comment:
Do you think it is possible to refactor all of these Loss part so that our model building function can be dtype unaware?